src/features/hostapp/hooks/target-hostapp.ts

```diff
@@ -202,58 +211,110 @@ async function getOSReleaseResource(
 						},
 					},
 				},
-				$and: [
+				$or: [
 					{
-						release_tag: {
-							$any: {
-								$alias: 'rt',
-								$expr: {
-									rt: {
-										value: normalizeOsVersion(osVersion),
-										tag_key: 'version',
-									},
-								},
-							},
-						},
+						// This effectively normalizes the versions string and
+						// compares it with the `semver` computed term, but we are
+						// using the individual fields so that DB indexes can also be used.
+						semver_major: parsedOsVersion.major,
+						semver_minor: parsedOsVersion.minor,
+						semver_patch: parsedOsVersion.patch,
+						semver_prerelease: parsedOsVersion.prerelease.join('.'),
+						semver_build: parsedOsVersion.build.join('.'),
+						revision: getRevisionFromSemver(parsedOsVersion),
```

Comment:
`getRevisionFromSemver()` defaults to the number `0` while the `revision` field in the db defaults to `null`. Is this not a problem?

Reply:
Not it isn't, since atm at least we only support provisioning w/ finalized OS releases.

Reply:
That was interesting actually... let me check what would mean to make the API support provisioning w/ raft OS releases right away :smiling_imp:

Reply:
Done :)